learning/step_by_step/scripting.rst

```diff
@@ -7,75 +7,86 @@ Introduction
+in this direction, adding tools that try to reduce the amount of code that needs
+to be written rather than eliminating it. The engine can provide many general
+solutions, while the developer can use code to accomplish specific tasks.
+
+Godot has embraced this goal from the beginning, and it has influenced many of
+its design decisions. First and foremost is the scene system. This system has
+many benefits, but fundamentally its goal is to relieve programmers from the
+responsibility of having to implement an overall architecture.
+
+As discussed in the previous section, when designing games using scenes, you
+will create many small scenes that can work together (in contrast to monolithic
+scenes that are kept separate from each other). We will give plenty more
+examples of this later on, but it's very important to remember: scenes are
+*complementary*.
 
 GDScript
 --------
 
-:ref:`doc_gdscript` is a dynamically typed scripting language to fit
-inside Godot. It was designed with the following goals:
+:ref:`doc_gdscript` is a dynamically typed scripting language which was
+designed with the following goals:
 
--  First and most importantly, making it simple, familiar and as easy to
-   learn as possible.
--  Making the code readable and error safe. The syntax is mostly
-   borrowed from Python.
+- Most importantly, it should feel simple, familiar, and as easy to learn as
+  possible.
+- It should have a syntax that's very readable. The syntax is mostly borrowed
+  from Python.
+- It should integrate tightly with Godot itself, for example sharing its memory
+  model, taking advantage of the scene/node system, and exposing useful
+  game-related classes already part of the Godot engine.
 
-Programmers generally take a few days to learn it, and within two weeks
-feel comfortable with it.
+Programmers generally take a few days to learn GDScript and feel comfortable
+with it within two weeks.
 
-As with most dynamically typed languages, the higher productivity
-(code is easier to learn, faster to write, no compilation, etc) is
-balanced with a performance penalty. But most critical code is written
-in C++ already in the engine (vector ops, physics, math, indexing, etc),
-resulting in a more than sufficient performance for most types of
-games.
+As with most dynamically typed languages, the higher productivity (code is
+easier to learn, faster to write, no compilation, etc.) is balanced with a
+performance penalty. However, keep in mind that most critical code is already
+written in C++ in the engine (vector ops, physics, math, indexing, etc.), which
+results in more than sufficient performance for most types of games.
 
 In any case, if more performance is required, critical sections can be
-rewritten in C++ and exposed transparently to the script. This allows
-the replacement of a GDScript class with a C++ class without altering
-the rest of the game.
+rewritten in C++ and registered with Godot, which in turn exposes them to all
```

Comment:
I assume this section was previously alluding to http://docs.godotengine.org/en/stable/development/cpp/custom_modules_in_cpp.html  However, it's now a bit more vague since Godot is introducing GDNative. Therefore, I tried to make it a bit clearer. Let me know if I missed the mark.